src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1282,8 +1282,14 @@ unsigned uct_dc_mlx5_ep_dci_release_progress(void *arg)
     UCS_STATIC_ASSERT((sizeof(iface->tx.dci_pool_release_bitmap) * 8) <=
                        UCT_DC_MLX5_IFACE_MAX_DCI_POOLS);
 
-    ucs_for_each_bit(pool_index, iface->tx.dci_pool_release_bitmap) {
-        ucs_assert(pool_index < iface->tx.num_dci_pools);
+    while (iface->tx.dci_pool_release_bitmap != 0) {
+        /* Take one DCI pool, and process all its released DCIs.
+         * It's possible that more DCIs to release will be added by the call to
+         * uct_dc_mlx5_iface_progress_pending() below, so we check the pool
+         * bitmap every time.
+         */
+        pool_index = ucs_ffs32(iface->tx.dci_pool_release_bitmap);
```

Comment:
maybe it is worth having assertion here as it was? Just to make sure that we don’t have some other pool indices in release bitmap
`ucs_assert(pool_index < iface->tx.num_dci_pools);`